cmake/modules/platform/arch/s390.cmake

```diff
@@ -0,0 +1,31 @@
+##############################################################################
+#
+# (c) Copyright IBM Corp. 2017,2017
+#
+#  This program and the accompanying materials are made available
+#  under the terms of the Eclipse Public License v1.0 and
+#  Apache License v2.0 which accompanies this distribution.
+#
+#      The Eclipse Public License is available at
+#      http://www.eclipse.org/legal/epl-v10.html
+#
+#      The Apache License v2.0 is available at
+#      http://www.opensource.org/licenses/apache2.0.php
+#
+# Contributors:
+#    Multiple authors (IBM Corp.) - initial implementation and documentation
+###############################################################################
+
+if (OMR_HOST_OS STREQUAL "linux")
+	if(OMR_ENV_DATA64)

```

Comment:
AFAICT `S39064` and `S390` flag is only used for Linux, maybe we should move this to the os/linux.cmake file.

In the future this should be renamed to `LINUXS390`

Reply:
You would have to check the arch but technically we do not know the arch at that point.  We do in fact know the arch at that point right now but I am not sure if that will stay the same.  Let me look at it a bit more.

Reply:
I am finding the distinction between OS and ARCH a bit difficult to make decisions in the cmake files.  I will think about it a bit more this afternoon.

You are correct though that these are purely  linux 390 flags.  It is a much bigger change to change them to be LINUXS390 and I believe we probably would not want to do that.  I think being able to check LINUX && S390 is better than just LINUX390.

Reply:
Ok moved